templates/config.hcl.j2

```diff
@@ -1,5 +1,16 @@
 pid_file = "/var/run/vault/vault.pid"
 
+{% if vault_web_ui %}
+ui = true
+{% endif %}
+
+{% if vault_api_addr is defined %}
```

Comment:
Is better to check if this is not False
```suggestion
{% if vault_api_addr %}
```

Reply:
Unfortunately that is dangerous. Having this playbook:

```
- hosts: all
  connection: local
  vars:
    test_ok: ""
    test_not_ok: "blah"

  tasks:
    - debug: msg="test_ok exists and empty"
      when: test_ok
      ignore_errors: yes
    - debug: msg="test_ok exists"
      when: test_ok is defined
      ignore_errors: yes
    - debug: msg="test_not_ok exists and empty"
      when: test_not_ok
      ignore_errors: yes
    - debug: msg="test_not_ok exists"
      when: test_not_ok is defined
      ignore_errors: yes
```

This is the output:

```

PLAY [all] ***********************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************
ok: [localhost]

TASK [debug] *********************************************************************************************************************************************
skipping: [localhost]

TASK [debug] *********************************************************************************************************************************************
ok: [localhost] => {
    "msg": "test_ok exists"
}

TASK [debug] *********************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'test_not_ok' failed. The error was: error while evaluating conditional (test_not_ok): 'blah' is undefined\n\nThe error appears to have been in '/home/jordi/Repos/stuart/infrastructure/ansible/a.test': line 14, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n      ignore_errors: yes\n    - debug: msg=\"test_not_ok exists and empty\"\n      ^ here\n"}
...ignoring

TASK [debug] *********************************************************************************************************************************************
ok: [localhost] => {
    "msg": "test_not_ok exists"
}

PLAY RECAP ***********************************************************************************************************************************************
localhost                  : ok=4    changed=0    unreachable=0    failed=0   
```

When a variable is empty string condition evaluates to false.
When a variable is empty, filter is defined evaluates to true.
When a variable is not empty string, it throws an error.
When a variable is not empty string, filter is defined evaluates to true.

The 3rd conditions is the problematic one, hence I wouldn't advice this change.

Reply:
I see and understand your point. Keep in mind this can't happen if the var is defined in the defaults.yml